### PR TITLE
Missing entries from demozoo

### PR DESCRIPTION
**Pull request by dag7dev**
What's been included:
- old roms that didn't scraped very well
- newer roms: since the script creation in 2021, newer productions have been added

The usual saulin is missing (it has been already removed in the past).

### CODE REVIEW

**Review by avivace — CHANGES_REQUESTED** [review]
Some referenced screenshots cannot be found? Could you try running the tests locally to see what breaks @dag7dev ?

File: entries/veitikka-ja-visionaari/game.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+            "filename": "veitikka-ja-visionaari.gb",
+            "playable": true
+        }
+    ],
+    "platform": "GB",
+    "screenshots": [
+        "veitikka-ja-visionaari.png"
+    ],
+    "slug": "veitikka-ja-visionaari",
+    "title": "Veitikka ja vision\u00e4\u00e4ri",
```
Comment (avivace) [inline entries/veitikka-ja-visionaari/game.json:16]: I believe we have encoding issues here..

**Comment by avivace** [comment]
@dag7dev I've also updated the scraper to retain the original demozoo link and parse the release date (when it's complete (dd-mm-yyyy)

**Comment by dag7dev** [comment]
> I believe we have encoding issues here..
is it an issue? they are these characters: ää

> 
@dag7dev I've also updated the scraper to retain the original demozoo link and parse the release date (when it's complete (dd-mm-yyyy)

isnt better yyyy-mm-dd ?

**Comment by avivace** [comment]
> > I believe we have encoding issues here..
> 
> is it an issue? they are these characters: ää
> 
> > @dag7dev I've also updated the scraper to retain the original demozoo link and parse the release date (when it's complete (dd-mm-yyyy)
> 
> isnt better yyyy-mm-dd ?

sorry yes that's the date that gets set. What I meant was that this happen when the full information is available (day, month year) in the "released" line in demozoo, but not when this is partial (e.g. "2009" or "august 2009" cases are not handled yet)

**Comment by dag7dev** [comment]
@avivace could we merge this and fix the importers later?

I plan to run them again as soon as I refactor everything, because everything needs to be tested.

**Comment by avivace** [comment]
The importer needs to be rerun to include the new modifications and this PR has to be updated accordingly. The link to the original release is very important and currently not included in the scraped entries

**Comment by avivace** [comment]
Rebased against master

**Comment by avivace** [comment]
Also, @dag7dev when I try to run the demozoo importer myself, no .gb rom is properly downloaded, but I get full HTML files...

**Comment by avivace** [comment]
@dag7dev I've merged the new entries in https://github.com/gbdev/database/commit/7c9dca0dda2e06a29051e86a242ff44d9e682620 and https://github.com/gbdev/database/commit/f003a0f84e4b9b55e4295a7b98d8aeb0dbda64ce, however, the scraper/this PR was not in the condition to be merged:

- Encoding is broken. There are a number of game.json with unicode entities escaped and written in the title. **Those still need to be fixed**
- There was no step checking for duplicates and comparing checksums against the ones we already have. This step flagged 1 duplicate entry
- Your PR didn't include all the new entries that I see running the script now
- Date parsing was missing. This is done. I worked on this and now we have dates, including imprecise ones (only year, year and month and full ones).
- Screenshots are still missing. Only one is taken and we don't navigate to the page containing all of them. We need a script now that can add them back, potentially to the old scraped entries as well.
- There are a number of duplicates which are not flagged by slug similarities and checksum collisions (20-y and pocket tangrams, from what I could see)
- Upstream link was missing, so we were effectively losing the 'source' of those entries. I've added this and we should have a demozoo link as website now.

**Comment by avivace** [comment]
Also, .gbc files are saved as .gb, for some reason. Not sure if you fixed those manually when committing here or

**Pull request closed without merging**